File: arrays/pop.c
#include <stdio.h>

int main()
{

  // initialize original array: numbers 1 - 10
  int arr[100] = { 0 };
  int i, x, pos, n = 10;

  // initial array size of 10
  for (i = 0; i < 10; i++)
  {
    arr[i] = i + 1;
  }

  // print the array now
  printf("original array is: ");
  for (i = 0; i < 10; i++)
  {
    printf("%d ", arr[i]);
  }
  printf("\n");

  /*--------------------------------------*/

  // removes element at position mentioned and returns it

  // get position, var pos
  pos = 5;

  // return element at that position
  printf("element at position %i is: %d\n", pos, arr[pos]);


  // move all elements in array back one at that position on
  for (i = pos; i < n - 1; i++)
  {
    arr[i] = arr[i + 1];
  }

  // print the new array
  printf("the new array is: ");
  for (i =0; i < n; i++)
  {
    printf("%d ", arr[i]);
  }
  printf("\n");

}

File: arrays/pop.py
import array

arr = array.array('i',[])

for i in range(1, 11):
    arr.append(i)

print('the original array is: ', end='')
for i in range(0, len(arr)):
    print(arr[i], end=' ')

print('\r')

# POP
# removes element at the position mentioned and returns it
print('popped element is: ', end='')
print(arr.pop(2))

print('the array after popping is: ', end='')
for i in range(0, len(arr)):
    print(arr[i], end=' ')

print('\r')
